src/runtime_src/xocl/api/xlnx/clPollStreams.cpp

```diff
@@ -45,10 +47,11 @@ clPollStreams(cl_device_id              device,
 	cl_int                          timeout,
 	cl_int*                         errcode_ret)
 {
+  int ret;
   validOrError(device,completions,min,max,actual,timeout,errcode_ret);
-  xocl::xocl(device)->poll_streams(completions,min,max,actual,timeout);
-  xocl::assign(errcode_ret,CL_SUCCESS);
-  return CL_SUCCESS;
+  ret = xocl::xocl(device)->poll_streams(completions,min,max,actual,timeout);
+  *errcode_ret = (ret < 0) ? ret : 0;
+  return (ret && ret != -ETIMEDOUT) ? CL_DEVICE_NOT_AVAILABLE : CL_SUCCESS;
```

Comment:
Please, no return codes.  Fix to throw appropriately at the right place.  I think device->poll_streams should throw.

Reply:
changed return CL_SUCCESS and throw error in poll_streams. User app might be check the *error_code but not error out, so I keep *error_code.